src/references/referenceService.ts

```diff
@@ -0,0 +1,230 @@
+import { IFormattedText, IReferenceInstance, IReference } from "./reference";
+
+export class referenceService {
```

Comment:
Class names should be [PascalCase](https://techterms.com/definition/pascalcase)

```suggestion
export class ReferenceService {
```